### DISCUSSION TITLE
[bug routing] In production direct url like domain.com/page redirection not working

### DISCUSSION THREAD

**Discussion by n1c01a5** [vercel/next.js#14010] [Help]
# Bug report

## Describe the bug

When I put in my browser an url like `mynextjswebsite.com/page` the application displays the homepage and not the `page` screen (only in production mode). But it's working if we click on a `Link` button with this route (/page).

For information, I use the static generation build of NextJS on a personnal server.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to 'example.com/page' directly in the url
2. See the application is set on the homepage not the page.

## Expected behavior

I want to go the page specified in the url and not on the homepage.

Thanks per advance,

**Comment by illuminist** [comment]
Does your server have any rule that's rewriting any request to root path(`/`)? Usually normal app that created from create-react-app need to rewriting a request to root path, but with nextjs SSG, the page can be request directly from path without rewritting.

**Comment by todortotev** [comment]
Please provide repository for reproduction.
